src/main/java/hudson/plugins/active_directory/ActiveDirectoryAuthenticationProvider.java

```diff
@@ -265,6 +268,50 @@ private boolean isAccountDisabled(IADsUser usr) {
         }
     }
 
+    private boolean isAccountExpired(IADsUser usr) {
+        try {
+            Date expirationDate = usr.accountExpirationDate();
+            if (expirationDate != null) {
+                return new Date().after(expirationDate);
+            }
+            return false;
+        } catch (ComException e) {
+            // keep the same behavior as in isAccountDisabled, not sure of the details as the linked page is now a 404
+            if (e.getHRESULT() == 0x8000500D) {
```

Comment:
Refactor into constant?

Reply:
Initially I did not want to touch existing code but as I was using multiple times the value, no problem with that. Thx for the point! I added some documentation on that particular case, that is not a 404.